File: CS/CustomMvvmFormWithRecurrenceExample/Model/MedicalAppointment.cs
using System;

namespace CustomMvvmFormWithRecurrenceExample.Model {
    public class MedicalAppointment {
        public MedicalAppointment(DateTime startTime, DateTime endTime, int doctorId, string notes, string location, string patientName, string insuranceNumber, bool firstVisit) {
            StartTime = startTime;
            EndTime = endTime;
            DoctorId = doctorId;
            Notes = notes;
            Location = location;
            PatientName = patientName;
            InsuranceNumber = insuranceNumber;
            FirstVisit = firstVisit;
        }

        protected MedicalAppointment() { }
        public int Id { get; set; }
        public bool AllDay { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public string PatientName { get; set; }
        public string Notes { get; set; }
        public string Subject { get; set; }
        public int PaymentStateId { get; set; }
        public int IssueId { get; set; }
        public int Type { get; set; }
        public string Location { get; set; }
        public string RecurrenceInfo { get; set; }
        public string ReminderInfo { get; set; }
        public int? DoctorId { get; set; }
        public string InsuranceNumber { get; set; }
        public bool FirstVisit { get; set; }
    }
}

File: VB/CustomMvvmFormWithRecurrenceExample/Model/MedicalAppointment.vb
Namespace CustomMvvmFormWithRecurrenceExample.Model

    Public Class MedicalAppointment

        Public Sub New(ByVal startTime As Date, ByVal endTime As Date, ByVal doctorId As Integer, ByVal notes As String, ByVal location As String, ByVal patientName As String, ByVal insuranceNumber As String, ByVal firstVisit As Boolean)
            Me.StartTime = startTime
            Me.EndTime = endTime
            Me.DoctorId = doctorId
            Me.Notes = notes
            Me.Location = location
            Me.PatientName = patientName
            Me.InsuranceNumber = insuranceNumber
            Me.FirstVisit = firstVisit
        End Sub

        Protected Sub New()
        End Sub

        Public Property Id As Integer

        Public Property AllDay As Boolean

        Public Property StartTime As Date

        Public Property EndTime As Date

        Public Property PatientName As String

        Public Property Notes As String

        Public Property Subject As String

        Public Property PaymentStateId As Integer

        Public Property IssueId As Integer

        Public Property Type As Integer

        Public Property Location As String

        Public Property RecurrenceInfo As String

        Public Property ReminderInfo As String

        Public Property DoctorId As Integer?

        Public Property InsuranceNumber As String

        Public Property FirstVisit As Boolean
    End Class
End Namespace
